implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -76,8 +85,37 @@ default DynatraceApiVersion apiVersion() {
                 .orElse(DynatraceApiVersion.V1);
     }
 
+    default String metricKeyPrefix() {
+        // returns an empty string if nothing is set in the properties.
+        return getString(this, "metricKeyPrefix").orElse("");
+    }
+
+    @Nullable
+    default Map<String, String> defaultDimensions() {
+        return Collections.emptyMap();
+    }
+
+    default Boolean enrichWithOneAgentMetadata() {
+        // defaults to false if nothing is set.
+        return getBoolean(this, "enrichWithOneAgentMetadata")
+                .orElse(false);
+    }
+
     @Override
     default Validated<?> validate() {
+        Validated<DynatraceApiVersion> checkApiVersion = checkRequired("apiVersion", DynatraceConfig::apiVersion).apply(this);
+        if (checkApiVersion.isInvalid()) {
+            return checkApiVersion;
+        }
+
+        if (apiVersion() == DynatraceApiVersion.V2) {
+            return checkAll(this,
+                    c -> StepRegistryConfig.validate(c),
+                    checkRequired("apiVersion", DynatraceConfig::apiVersion),
+                    check("apiToken", DynatraceConfig::apiToken)
+                            .andThen(v -> v.invalidateWhen(x -> checkRequired("uri", DynatraceConfig::uri).apply(this).isInvalid(), "when using an API token, the endpoint URI is required", InvalidReason.MISSING))
+            );
+        }
         return checkAll(this,
```

Comment:
I think an if/else or switch/case would be cleaner here than relying on an early return.

Reply:
changed.